File: Mathématiques/5 – Algorithmique du baccalauréat/5 - Convergence de suites adjacentes.py
from math import *
a = float(input())
b = float(input())

while round(a, 5) != round(b, 5):
    _a = (a+b)/2
    b = sqrt((a ** 2 + b ** 2)/2)
    a = _a

print(round(a, 5))
print(round(b, 5))

File: Mathématiques/5 – Algorithmique du baccalauréat/5 - Convergence de suites adjacentes.cpp
#include <iostream>
#include <cmath>
using namespace std;

double round5(double n)
{
  return round(n * 100000) / 100000;
}

int main()
{
  double a, b;
  cin >> a >> b;
  while (round5(a) != round5(b))
  {
    double _a = (a + b) / 2;
    b = sqrt((pow(a, 2) + pow(b, 2)) / 2);
    a = _a;
  }
  cout << round5(a) << endl
       << round5(b);
}